app/decorators/user_decorator.rb

```diff
@@ -1,9 +1,9 @@
 class UserDecorator < ApplicationDecorator
   delegate_all
 
-  def proposal_url(proposal)
+  def proposal_url(proposal, force_public_url=false)
     event = proposal.event
-    if model.staff_for? event
+    if model.staff_for?(event) && !force_public_url
```

Comment:
Oh now I remember. In public_comment#notify, we have access to the user who left the comment, but once you're inside notification#create_for, you only have the users who are the target of the notification. Also, it's #create_for that gets used in different spots. The flag was a hacky way to make sure the other uses of that method weren't affected. I'll see if that's still an issue with your suggestion; should only take a few minutes